consensus/src/sync/live/block_queue/block_request_component.rs

```diff
@@ -170,55 +170,72 @@ impl<N: Network> BlockRequestComponent<N> {
                     // 3. Check that hash chain verifies
                     // We need to pass through the validators once we reach a new epoch
                     // to be able to verify macro blocks
-                    let blocks = &response.blocks;
+                    {
```

Comment:
Does not work since we are taking twice by reference the first and last blocks and storing them in temporary variables, we need to drop these before the `&mut` can be taken.

Reply:
For completion purposes, it is possible to remove the scopes because rust has Non-lexical lifetimes which allows the borrows to be dropped early if they are no longer used, thus allowing for other borrows to take place.